crypto/src/vrf/transcript.rs

```diff
@@ -0,0 +1,108 @@
+// Copyright (c) 2021 RBB S.r.l
+// opensource@mintlayer.org
+// SPDX-License-Identifier: MIT
+// Licensed under the MIT License;
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	http://spdx.org/licenses/MIT
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// Author(s): S. Afach
+
+use merlin::Transcript;
+
+#[derive(Clone, Debug, Eq, PartialEq)]
+pub enum TranscriptComponent {
+    RawData(Vec<u8>),
+    U64(u64),
+}
+
+#[derive(Clone, Debug, Eq, PartialEq)]
+pub struct TranscriptAssembler {
+    label: &'static [u8],
+    components: Vec<(&'static [u8], TranscriptComponent)>,
+}
+
+// A wrapper that makes it unnecessary to directly use the merlin dependency
+#[derive(Clone)]
+pub struct WrappedTranscript(Transcript);
```

Comment:
Maybe call this just `Transacript` and refer to the `Transcript` type from the upstream crate by the full name, i.e. `merlin::Transcript`. I'd prefer our code base to be free of filler words like `Wrapper`.

Reply:
I've had a similar discussion with stanislav today about `Config` vs `ChainConfig` vs `ChainstateConfig`. The problem is that IDEs automatically add `use` clauses and we're gonna have to depend on the discretion of the programmer to make the reader not have two different Transcript or Config contexts in the same file in a readable way for symbol names that touch multiple crates. I'd rather be a little more verbose than ambiguous for the reader. After all, the rust code is written for humans, not for the machine. The machine couldn't care less.